index.js

```diff
@@ -0,0 +1,352 @@
+     * @method _updateCommitStatus
+     * @param  {Object}     config              Configuration
+     * @param  {String}     config.scmContext   Name of scm context
+     * @return {Promise}
+     */
+    _updateCommitStatus(config) {
+        return this.chooseScm(config).then(scm => scm.updateCommitStatus(config));
+    }
+
+    /**
+     * Fetch content of a file from an scm repo
+     * @method _getFile
+     * @param  {Object}     config              Configuration
+     * @param  {String}     config.scmContext   Name of scm context
+     * @return {Promise}                        content of a scmContext file
+     */
+    _getFile(config) {
+        return this.chooseScm(config).then(scm => scm.getFile(config));
+    }
+
+    /**
+     * Get list of objects which consists of opened PR names and its ref
+     * @method _getOpenedPRs
+     * @param  {Object}     config              Configuration
+     * @param  {String}     config.scmContext   Name of scm context
+     * @return {Promise}                        Opened PRs of scmContext
+     */
+    _getOpenedPRs(config) {
+        return this.chooseScm(config).then(scm => scm.getOpenedPRs(config));
+    }
+
+    /**
+     * Return a valid Bell configuration of all registered modules (for OAuth)
+     * @method _getBellConfiguration
+     * @return {Promise}
+     */
+    _getBellConfiguration() {
+        return this.allScm(scm => scm.getBellConfiguration());
+    }
+
+    /**
+     * Resolve a pull request object based on the config
+     * @method _getPrInfo
+     * @param  {Object}     config              Configuration
+     * @param  {String}     config.scmContext   Name of scm context
+     * @return {Promise}                        PR info of scmContext
+     */
+    _getPrInfo(config) {
+        return this.chooseScm(config).then(scm => scm.getPrInfo(config));
+    }
+
+    /**
+     * Return statistics on the scm of all registered modules
+     * @method stats
+     * @return {Object} object           Hash containing metrics for the scm
+     */
+    stats() {
+        let result = {};
+
+        Object.keys(this.scms).forEach((key) => {
+            result = Object.assign(result, this.scms[key].stats());
+        });
+
+        return result;
+    }
+
+    /**
+     * Get an array of scm context (e.g. [github:github.com, gitlab:mygitlab])
+     * @method _getScmContexts
+     * @return {Array}
+     */
+    _getScmContexts() {
+        return Object.keys(this.scms);
+    }
+
+    /**
+     * Determine a scm module can handle the received webhook
+     * @method _canHandleWebhook
+     * @param  {Object}     headers     The request headers associated with the webhook payload
+     * @param  {Object}     payload     The webhook payload received from the SCM service
+     * @return {Promise}
+     */
+    _canHandleWebhook(headers, payload) {
+        return this.chooseWebhookScm(headers, payload)
+            .then(scm => scm.canHandleWebhook(headers, payload));
```

Comment:
I think it's because the purpose of `chooseWebhookScm` is to return an `scm`. So after it chooses the correct scm, it needs to call `canHandleWebhook` on that specific scm. 

`chooseWebhookScm` is also used by `parseHook`, so I guess that's why it needed to return the actual `scm`.